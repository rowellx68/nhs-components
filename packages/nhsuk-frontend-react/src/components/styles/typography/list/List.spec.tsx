import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Default as DefaultStory,
  Bulleted as BulletedStory,
  Numbered as NumberedStory,
  WithBorders as WithBordersStory,
} from './List.stories';

const Default = composeStory(DefaultStory, meta);
const Bulleted = composeStory(BulletedStory, meta);
const Numbered = composeStory(NumberedStory, meta);
const WithBorders = composeStory(WithBordersStory, meta);

it('should render the List component', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('.nhsuk-list')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the List component with bulleted items', () => {
  const { container } = render(<Bulleted />);

  expect(container.querySelector('ul')).toBeInTheDocument();
  expect(container.querySelector('.nhsuk-list--bullet')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the List component with numbered items', () => {
  const { container } = render(<Numbered />);

  expect(container.querySelector('ol')).toBeInTheDocument();
  expect(container.querySelector('.nhsuk-list--number')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the List component with borders', () => {
  const { container } = render(<WithBorders />);

  expect(container.querySelector('.nhsuk-list--border')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
