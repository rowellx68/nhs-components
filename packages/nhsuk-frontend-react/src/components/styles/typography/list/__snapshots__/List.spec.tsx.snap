// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the List component 1`] = `
<div>
  <ul
    class="nhsuk-list"
  >
    <li>
      <a
        href="#"
      >
        Money, work and benefits
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Care after a hospital stay
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Support and benefits for carers
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render the List component with borders 1`] = `
<div>
  <ul
    class="nhsuk-list nhsuk-list--bullet nhsuk-list--border"
  >
    <li>
      Money, work and benefits
    </li>
    <li>
      Care after a hospital stay
    </li>
    <li>
      Support and benefits for carers
    </li>
  </ul>
</div>
`;

exports[`should render the List component with bulleted items 1`] = `
<div>
  <p>
    Symptoms can include:
  </p>
  <ul
    class="nhsuk-list nhsuk-list--bullet"
  >
    <li>
      tiredness and lack of energy
    </li>
    <li>
      shortness of breath
    </li>
    <li>
      noticeable heartbeats (heart palpitations)
    </li>
    <li>
      pale skin
    </li>
  </ul>
</div>
`;

exports[`should render the List component with numbered items 1`] = `
<div>
  <h3
    class=""
  >
    How to gargle with salt water
  </h3>
  <ol
    class="nhsuk-list nhsuk-list--number"
  >
    <li>
      Dissolve half a teaspoon of salt in a glass of warm water.
    </li>
    <li>
      Gargle with the solution then spit it out â€“ don't swallow it.
    </li>
    <li>
      Repeat as often as you like.
    </li>
  </ol>
</div>
`;
